docs/book/src/user/quick-start.md

```diff
@@ -0,0 +1,477 @@
+    controllerManager:
+      extraArgs:
+        cloud-provider: aws
+```
+{{#/tab }}
+{{#tab vSphere}}
+
+<aside class="note warning">
+
+<h1>Action Required</h1>
+
+These examples include environment variables that you should substitute before creating the resources.
+
+</aside>
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha2
+kind: Machine
+metadata:
+  name: capi-quickstart-controlplane-0
+  labels:
+    cluster.x-k8s.io/control-plane: "true"
+    cluster.x-k8s.io/cluster-name: "capi-quickstart"
+spec:
+  version: v1.15.3
+  bootstrap:
+    configRef:
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+      kind: KubeadmConfig
+      name: capi-quickstart-controlplane-0
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    kind: VSphereMachine
+    name: capi-quickstart-controlplane-0
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+kind: VSphereMachine
+metadata:
+  name: capi-quickstart-controlplane-0
+spec:
+  datacenter: "${VSPHERE_DATACENTER}"
+  network:
+    devices:
+    - networkName: "${VSPHERE_NETWORK}"
+      dhcp4: true
+      dhcp6: false
+  numCPUs: ${VSPHERE_NUM_CPUS}
+  memoryMiB: ${VSPHERE_MEM_MIB}
+  diskGiB: ${VSPHERE_DISK_GIB}
+  template: "${VSPHERE_TEMPLATE}"
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+kind: KubeadmConfig
+metadata:
+  name: capi-quickstart-controlplane-0
+spec:
+  initConfiguration:
+    nodeRegistration:
+      name: "{{ ds.meta_data.hostname }}"
+      criSocket: "/var/run/containerd/containerd.sock"
+      kubeletExtraArgs:
+        cloud-provider: vsphere
+  clusterConfiguration:
+    apiServer:
+      extraArgs:
+        cloud-provider: vsphere
+        cloud-config: /etc/kubernetes/vsphere.conf
+      extraVolumes:
+      - name: "cloud-config"
+        hostPath: /etc/kubernetes/vsphere.conf
+        mountPath: /etc/kubernetes/vsphere.conf
+        readOnly: true
+        pathType: File
+    controllerManager:
+      extraArgs:
+        cloud-provider: vsphere
+        cloud-config: /etc/kubernetes/vsphere.conf
+      extraVolumes:
+      - name: "cloud-config"
+        hostPath: /etc/kubernetes/vsphere.conf
+        mountPath: /etc/kubernetes/vsphere.conf
+        readOnly: true
+        pathType: File
+  files:
+  - path: /etc/kubernetes/vsphere.conf
+    owner: root:root
+    permissions: "0600"
+    encoding: base64
+    content: |
+      ${CLOUD_CONFIG_B64ENCODED}
+  users:
+  - name: capv
+    sudo: "ALL=(ALL) NOPASSWD:ALL"
+    sshAuthorizedKeys:
+    - "${SSH_AUTHORIZED_KEY}"
+  preKubeadmCommands:
+  - hostname "{{ ds.meta_data.hostname }}"
+  - echo "::1         ipv6-localhost ipv6-loopback" >/etc/hosts
+  - echo "127.0.0.1   localhost {{ ds.meta_data.hostname }}" >>/etc/hosts
+  - echo "{{ ds.meta_data.hostname }}" >/etc/hostname
+```
+{{#/tab }}
+{{#/tabs }}
+
+After the controlplane is up and running, we should apply the addons to install Calico.
+
+{{#tabs name:"tab-usage-addons" tabs:"AWS,vSphere"}}
```

Comment:
These tabs can be removed and we can just deploy the calico manifests directly:
```
kubectl apply -f https://docs.projectcalico.org/v3.8/manifests/calico.yaml
```

Reply:
Just realized, to apply calico, you'll need to download and extract the kubeconfig from the secret first:
```sh
kubectl get secret capi-quickstart -o json | jq -r .data.value | base64 -d > qs.kubeconfig
kubectl --kubeconfig qs.kubeconfig apply -f https://docs.projectcalico.org/v3.8/manifests/calico.yaml
```